src/providers/google/youtube-post.ts

```diff
@@ -0,0 +1,170 @@
+            // No results found, so stop sync
+            syncPosition = this.stopSync(syncPosition);
+
+            return {
+                position: syncPosition,
+                results: [],
+            };
+        }
+
+        const results = await this.buildResults(
+            youtube,
+            serverResponse,
+            syncPosition.breakId,
+            _.has(this.config, "metadata.breakTimestamp")
+                ? this.config.metadata.breakTimestamp
+                : undefined
+        );
+
+        syncPosition = this.setNextPosition(syncPosition, serverResponse);
+
+        if (results.length != this.config.batchSize) {
+            // Not a full page of results, so stop sync
+            syncPosition = this.stopSync(syncPosition);
+        }
+
+        return {
+            results,
+            position: syncPosition,
+        };
+    }
+
+    protected stopSync(syncPosition: SyncHandlerPosition): SyncHandlerPosition {
+        if (syncPosition.status == SyncHandlerStatus.STOPPED) {
+            return syncPosition;
+        }
+
+        syncPosition.status = SyncHandlerStatus.STOPPED;
+        syncPosition.thisRef = undefined;
+        syncPosition.breakId = syncPosition.futureBreakId;
+        syncPosition.futureBreakId = undefined;
+
+        return syncPosition;
+    }
+
+    protected setNextPosition(
+        syncPosition: SyncHandlerPosition,
+        serverResponse: GaxiosResponse<youtube_v3.Schema$ActivityListResponse>
+    ): SyncHandlerPosition {
+        if (!syncPosition.futureBreakId && serverResponse.data.items.length) {
+            syncPosition.futureBreakId = `${this.connection.profile.id}-${serverResponse.data.items[0].id}`;
+        }
+
+        if (_.has(serverResponse, "data.nextPageToken")) {
+            // Have more results, so set the next page ready for the next request
+            syncPosition.thisRef = serverResponse.data.nextPageToken;
+        } else {
+            syncPosition = this.stopSync(syncPosition);
+        }
+
+        return syncPosition;
+    }
+
+    protected async buildResults(
+        youtube: youtube_v3.Youtube,
+        serverResponse: GaxiosResponse<youtube_v3.Schema$ActivityListResponse>,
+        breakId: string,
+        breakTimestamp?: string
+    ): Promise<SchemaPost[]> {
+        const results: SchemaPost[] = [];
+        for (const item of serverResponse.data.items) {
+            const itemId = `${this.connection.profile.id}-${item.id}`;
+
+            if (itemId == breakId) {
+                break;
+            }
+
+            const snippet = item.snippet;
+            const contentDetails = item.contentDetails;
+            const insertedAt = snippet.publishedAt || "Unknown";
+
+            if (breakTimestamp && insertedAt < breakTimestamp) {
+                break;
+            }
+
+            const title = snippet.title || "No title";
+
+            results.push({
+                _id: `youtube-post-${itemId}`,
+                name: title,
```

Comment:
See comments on youtube following

Reply:
Can you provide an example of all the data available in `item`? I'd like to see if there's more data we can include.

Reply:
> Can you provide an example of all the data available in `item`? I'd like to see if there's more data we can include.

### Youtube activity has multiple types and typical types are:
- upload: Indicates that a user has uploaded a video.
- like: Indicates that a user has liked a video.
- comment: Indicates that a user has commented on a video.
- subscription: Indicates that a user has subscribed to a channel.
- recommendation: Indicates that a user has recommended a video.
- playlistItem: Indicates that a user has added a video to a playlist.

Each type has different `contentDeatils` like this:
```
{
  "snippet": {
    "type": "upload"|"like"|"xxx",
    "channelId": "",
    "title": ",
    "description": "",
    "thumbnails": {
      "default": {
        "url": "https://example.com/default.jpg",
        "width": 120,
        "height": 90
      }
    }
  },
  "contentDetails": {
    "upload": {
      "videoId": "video_id"
    }
    "like": {
      "resourceId": {
        "kind": "youtube#video",
        "videoId": "liked_video_id"
      }
    }
  "xxx": {}
  }
}
```
### Youtube subscription response is static 
```
{
  "kind": "youtube#subscriptionListResponse",
  "etag": "etag",
  "items": [
    {
      "kind": "youtube#subscription",
      "etag": "etag",
      "id": "subscriptionId",
      "snippet": {
        "publishedAt": "2023-07-14T12:34:56.000Z",
        "title": "Channel Title",
        "description": "Channel Description",
        "resourceId": {
          "kind": "youtube#channel",
          "channelId": "channelId"
        },
        "channelId": "subscribedChannelId",
        "thumbnails": {
          "default": {
            "url": "https://example.com/default.jpg",
            "width": 88,
            "height": 88
          },
          "medium": {
            "url": "https://example.com/medium.jpg",
            "width": 240,
            "height": 240
          },
          "high": {
            "url": "https://example.com/high.jpg",
            "width": 800,
            "height": 800
          }
        }
      }
    }
  ]
}
```

Reply:
Thanks, we will need to make some adjustments here.

upload: ignore for now
like: favourite (schema doesn't yet exist)
comment: ignore for now
subscription: following
recommendation: favourite (schema doesn't yet exist)
playlistItem: ignore for now

I'll create the favourite schema and share with you today.

Reply:
> Thanks, we will need to make some adjustments here.
> 
> upload: ignore for now like: favourite (schema doesn't yet exist) comment: ignore for now subscription: following recommendation: favourite (schema doesn't yet exist) playlistItem: ignore for now
> 
> I'll create the favourite schema and share with you today.

Okay

Reply:
@chime3 See https://common.schemas.verida.io/favourite/v0.1.0/schema.json